inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_broadcast.cpp

```diff
@@ -29,19 +31,66 @@ void dynamicToStaticShapeBroadcast(std::shared_ptr<ngraph::Node> target) {
     }
 
-    auto dsr = std::make_shared<ngraph::vpu::op::DynamicShapeResolver>(
-            staticShapeBroadcast->output(0), broadcast->input_value(1));
-    dsr->set_friendly_name(broadcast->get_friendly_name());
+    std::shared_ptr<ngraph::Node> dsr;
+
+    if (broadcast->get_broadcast_spec() == ngraph::op::BroadcastType::BIDIRECTIONAL) {
+        const auto inputShape = broadcast->get_input_partial_shape(0);
+
+        const auto targetShape = broadcast->input_value(1).get_node_shared_ptr();
+        const auto shapeType = targetShape->get_element_type();
+
+        const auto inputShapeDimsCount = inputShape.rank().get_length();
+        const auto targetShapeDimsCount = ngraph::shape_size(broadcast->get_input_partial_shape(1).get_shape());
+
+        const auto inputShapeConst = std::make_shared<ngraph::opset5::Constant>(shapeType,
+                                                                                ngraph::Shape{static_cast<size_t>(inputShapeDimsCount)},
+                                                                                inputShape.get_shape());
+
+        const auto maxRankNode = inputShapeDimsCount > targetShapeDimsCount ? inputShapeConst : targetShape;
+        const auto minRankNode = maxRankNode == inputShapeConst ? targetShape : inputShapeConst;
+        const auto maxRank = maxRankNode == inputShapeConst ? inputShapeDimsCount : targetShapeDimsCount;
+        const auto minRank = minRankNode == inputShapeConst ? inputShapeDimsCount : targetShapeDimsCount;
+
+        ngraph::NodeVector dims;
+        for (int i = 0; i < minRank; i++) {
+            const auto minRankDim = std::make_shared<ngraph::opset5::Gather>(minRankNode,
+                                                                             ngraph::opset5::Constant::create(shapeType, ngraph::Shape{1}, {minRank - i - 1}),
+                                                                             ngraph::opset5::Constant::create(shapeType, ngraph::Shape{1}, {0}));
+            const auto maxRankDim = std::make_shared<ngraph::opset5::Gather>(maxRankNode,
+                                                                             ngraph::opset5::Constant::create(shapeType, ngraph::Shape{1}, {maxRank - i - 1}),
+                                                                             ngraph::opset5::Constant::create(shapeType, ngraph::Shape{1}, {0}));
+            dims.push_back(std::make_shared<ngraph::opset5::Maximum>(minRankDim, maxRankDim));
+        }
 
+        for (int i = maxRank - minRank - 1; i >= 0; i--) {
+            dims.push_back(std::make_shared<ngraph::opset5::Gather>(maxRankNode,
+                                                                    ngraph::opset5::Constant::create(shapeType, ngraph::Shape{1}, {i}),
+                                                                    ngraph::opset5::Constant::create(shapeType, ngraph::Shape{1}, {0})));
+        }
```

Comment:
I think all of this is better to be extracted to a separate helper with `revert` in name - if I got your idea correctly

Reply:
Explained below.